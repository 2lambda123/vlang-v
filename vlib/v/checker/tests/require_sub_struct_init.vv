module main

struct Abc {
	field int [required]
}

struct NestedAbc {
	Abc
}

struct FieldAbc {
	field Abc
}

fn main() {
	_ := NestedAbc{}
	_ := FieldAbc{}
}
