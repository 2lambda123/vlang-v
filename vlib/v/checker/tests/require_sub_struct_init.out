vlib/v/checker/tests/require_sub_struct_init.vv:16:7: error: field `NestedAbc.Abc.field` must be initialized
   14 | 
   15 | fn main() {
   16 |     _ := NestedAbc{}
      |          ~~~~~~~~~~~
   17 |     _ := FieldAbc{}
   18 | }
vlib/v/checker/tests/require_sub_struct_init.vv:17:7: error: field `FieldAbc.field.field` must be initialized
   15 | fn main() {
   16 |     _ := NestedAbc{}
   17 |     _ := FieldAbc{}
      |          ~~~~~~~~~~
   18 | }
   